src/main/java/org/elasticsearch/search/fetch/analyzedtext/AnalyzedTextRequest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.search.fetch.analyzedtext;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.support.single.shard.SingleShardRequest;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * A request to analyze a text associated with a specific index. Allow to provide
+ * the actual analyzer name to perform the analysis with.
+ */
+public class AnalyzedTextRequest extends SingleShardRequest<AnalyzedTextRequest> {
+
```

Comment:
This is a copy of what AnalyzeRequest used to be. The analyze request now supports specifying a complete analyzer definition (as settings) instead of the name with a lot of code to handle it which cannot be easily re-used inside AnalyzeAction. So, I basically see 3 choices here - implement support for specifying an analyzer, continue using it without support for the specified analyzer or create a specific request that doesn't support specifying an analyzer. I went with the 3rd option since it looked like a slightly cleaner solution. But I am fine with other 2 options as well.